pandas/core/arrays/base.py

```diff
@@ -283,10 +283,15 @@ def astype(self, dtype, copy=True):
         return np.array(self, dtype=dtype, copy=copy)
 
     def isna(self):
-        # type: () -> np.ndarray
-        """Boolean NumPy array indicating if each value is missing.
+        # type: () -> Union[ExtensionArray, np.ndarray]
+        """
+        An array indicating if each value is missing.
+
+        This should return a 1-D array the same length as `self`. This array
+        may be an ndarray or an ExtensionArray of the same type as `self`.
```

Comment:
> The "the same type as self", is that required?

Not especially, though remember that a hypothetical `ArrowArray` would probably support multiple **d**types. So a ``ArrowArray[int].isna()` could return an `ArrowArray[bool]`. I'll remove it.

Reply:
It's becoming off topic :), but I think it would be not unlikely that a arrow based system would still have BooleanArray, Float64Array, DatetimeArray, ..